File: itx-go/itx-compute-worker/internal/dispatcher/dispatcher.go
package dispatcher

import (
	"context"
	"log/slog"

	"github.com/chehsunliu/itx/itx-go/itx-compute-worker/internal/state"
)

type Dispatcher struct {
	//nolint:unused // wired up by upcoming handlers
	state state.WorkerState
}

func New(state state.WorkerState) *Dispatcher {
	return &Dispatcher{state: state}
}

func (d *Dispatcher) Handle(_ context.Context, body string) error {
	slog.Info("compute message received (no handler yet)", "body", body)
	return nil
}

File: itx-rs/itx-compute-worker/src/dispatcher.rs
use async_trait::async_trait;
use itx_contract::queue::{HandlerError, MessageHandler};

use crate::state::ComputeWorkerState;

pub struct ComputeDispatcher {
    #[allow(dead_code)] // wired up by upcoming handlers (e.g. send-notification → email API)
    state: ComputeWorkerState,
}

impl ComputeDispatcher {
    pub fn new(state: ComputeWorkerState) -> Self {
        Self { state }
    }
}

#[async_trait]
impl MessageHandler for ComputeDispatcher {
    async fn handle(&self, body: &str) -> Result<(), HandlerError> {
        tracing::info!(body = body, "compute message received (no handler yet)");
        Ok(())
    }
}
